_docs/contracts.md

```diff
@@ -113,6 +113,11 @@ Restrictions for libraries in comparison to contracts:
 
 (these might be lifted at a later point)
 
+Common pitfalls for libraries
+
+* The value for `msg.sender` will be that of the contract which is calling the library function.  (If A calls contract B which internally calls library C, then within the function call of library C, `msg.sender` will be the address of contract B).
+* When declaring local variables for structs or arrays, modifications to them will not persist in storage unless they are declared with `storage` as their storage location.  (For example, `var my_struct = self.some_mapping[key]` any changes to `my_struct` would not be persisted in contract storage).
```

Comment:
I would say this should rather go to the "reference types" section, but we can add a link here.

What would fit better here would be to say that when handing over a memory type to a library function, modifications will not be carried back to the value in the context of the caller. The reason is the same as above.